include/swift/AST/CanTypeVisitor.h

```diff
@@ -61,11 +61,11 @@ class CanTypeVisitor {
 #define TYPE(CLASS, PARENT) ABSTRACT_TYPE(CLASS, PARENT)
 #define ABSTRACT_SUGARED_TYPE(CLASS, PARENT)
 #define SUGARED_TYPE(CLASS, PARENT)
-  // Don't allow unchecked types by default, but allow visitors to opt-in to
+  // Don't allow invalid types by default, but allow visitors to opt-in to
   // handling them.
-#define UNCHECKED_TYPE(CLASS, PARENT)                          \
+#define INVALID_TYPE(CLASS, PARENT)                            \
```

Comment:
Sure, something like `TYPECHECKER_ONLY_TYPE` would work too. That being said, if special casing ErrorType is important, then I don't think we should rename "UNCHECKED_TYPE". Doing so would just be a distraction.

Separably, what reasonable thing should type visitor clients do when they find an error type? As I grep through the code right, nearly all cases of visitErrorType that aren't dumping/printing/walking related terminate with `llvm_unreachable`. What problem are we solving by forcing people to explicitly call `llvm_unreachable` rather than implicitly call it via the visitor?

Reply:
Well, you don't always want to crash upon seeing an ErrorType. For example both SIL and IRGen type lowering accept it. As I said, this is needed to avoid crashing on invalid code. There are surely existing visitors in there that currently crash on an ErrorType which should handle it some other way. This is why I suggested surfacing the ErrorType case explicitly in the visitor.